internal/charmstore/migrations_test.go

```diff
@@ -793,6 +831,692 @@ func (s *migrationsSuite) TestPopulateWriteACLSomeUpdates(c *gc.C) {
 	})
 }
 
+func (s *migrationsSuite) TestPopulatePromulgatedEntities(c *gc.C) {
+	s.patchMigrations(c, getMigrations("populate promulgated entities"))
+	id1 := charm.MustParseReference("trusty/django-42")
+	id2 := charm.MustParseReference("~who/utopic/rails-47")
+	id3 := charm.MustParseReference("~who/trusty/django-42")
+	id4 := charm.MustParseReference("~dalek/trusty/django-42")
+	id5 := charm.MustParseReference("~ace/utopic/rails-47")
+	id6 := charm.MustParseReference("~who/trusty/django-41")
+	err := s.db.Entities().Insert(&entity2{
+		URL:      id1,
+		BaseURL:  baseURL(id1),
+		BlobHash: "django-1",
+		Name:     id1.Name,
+		User:     id1.User,
+		Series:   id1.Series,
+		Revision: id1.Revision,
+		ExtraInfo: map[string][]byte{
+			"bzr-owner": []byte(`"who"`),
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	err = s.db.Entities().Insert(&entity2{
+		URL:      id2,
+		BaseURL:  baseURL(id2),
+		BlobHash: "rails-1",
+		Name:     id2.Name,
+		User:     id2.User,
+		Series:   id2.Series,
+		Revision: id2.Revision,
+		ExtraInfo: map[string][]byte{
+			"bzr-owner": []byte(`"who"`),
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	err = s.db.Entities().Insert(&entity2{
+		URL:      id3,
+		BaseURL:  baseURL(id3),
+		BlobHash: "django-1",
+		Name:     id3.Name,
+		User:     id3.User,
+		Series:   id3.Series,
+		Revision: id3.Revision,
+		ExtraInfo: map[string][]byte{
+			"bzr-owner": []byte(`"who"`),
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	err = s.db.Entities().Insert(&entity2{
+		URL:      id4,
+		BaseURL:  baseURL(id4),
+		BlobHash: "django-1",
+		Name:     id4.Name,
+		User:     id4.User,
+		Series:   id4.Series,
+		Revision: id4.Revision,
+		ExtraInfo: map[string][]byte{
+			"bzr-owner": []byte(`"dalek"`),
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	err = s.db.Entities().Insert(&entity2{
+		URL:      id5,
+		BaseURL:  baseURL(id5),
+		BlobHash: "rails-1",
+		Name:     id5.Name,
+		User:     id5.User,
+		Series:   id5.Series,
+		Revision: id5.Revision,
+		ExtraInfo: map[string][]byte{
+			"bzr-owner": []byte(`"ace"`),
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	err = s.db.Entities().Insert(&entity2{
+		URL:      id6,
+		BaseURL:  baseURL(id6),
+		BlobHash: "django-2",
+		Name:     id6.Name,
+		User:     id6.User,
+		Series:   id6.Series,
+		Revision: id6.Revision,
+		ExtraInfo: map[string][]byte{
+			"bzr-owner": []byte(`"who"`),
+		},
+	})
+	c.Assert(err, gc.IsNil)
+
+	s.insertBaseEntity(c, baseURL(id1), &mongodoc.ACL{})
+	s.insertBaseEntity(c, baseURL(id2), &mongodoc.ACL{})
+	s.insertBaseEntity(c, baseURL(id3), &mongodoc.ACL{})
+	s.insertBaseEntity(c, baseURL(id4), &mongodoc.ACL{})
+	s.insertBaseEntity(c, baseURL(id5), &mongodoc.ACL{})
```

Comment:
I suspect if you change insertBaseEntity to unset the promulgated field, then the tests will fail.